src/routes/GroupExplorer/GroupSearch/GroupSearch.test.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+import { unmountComponentAtNode } from 'react-dom'
+import { graphql } from 'msw'
+import { setupServer } from 'msw/node'
+import { history } from '../../../router'
+import { generateStore, render, AllTheProviders, waitFor, fireEvent } from 'util/reactTestingLibraryExtended'
+import GroupSearch from './GroupSearch'
+import orm from '../../../store/models'
+
+let container = null
+let providersWithStore = null
+
+export const handlers = [
+  graphql.operation((req, res, ctx) => {
+    const { search } = req.body.variables
+    let items = search !== 'different group'
+      ? [
+        {
+          accessibility: [3],
+          memberCount: 12,
+          description: 'Words do not belong in this here town',
+          location: 'Baltimore',
+          locationObject: {
+            city: 'Baltimore',
+            country: 'USA',
+            fullText: 'Baltimore, USA',
+            locality: '',
+            neighborhood: '',
+            region: 'East Coast'
+          },
+          id: '345',
+          avatarUrl: 'wee.com',
+          bannerUrl: 'wee.com',
+          name: 'Test Group Title',
+          slug: 'test-group-title',
+          groupTopics: [],
+          members: []
+        }
+      ]
+      : [
+        {
+          accessibility: [3],
+          memberCount: 16,
+          description: 'Completely different group don\'t you think',
+          location: 'Somewhere else',
+          locationObject: {
+            city: 'Austin',
+            country: 'USA',
+            fullText: 'Austin, USA',
+            locality: '',
+            neighborhood: '',
+            region: 'East Coast'
+          },
+          id: '345',
+          avatarUrl: 'wee.com',
+          bannerUrl: 'wee.com',
+          name: 'Search input results',
+          slug: 'test-group-title',
+          groupTopics: [],
+          members: []
+        }
+      ]
+    return res(
+      ctx.data({ groups: { hasMore: false, items, total: 0 } })
+    )
+  })
+]
+
+jest.mock('components/ScrollListener', () => () => <div />) // was throwing errors with this.element().removeEventListener('blabadlbakdbfl')
+
+const server = setupServer(...handlers)
+
+// Enable API mocking before tests.
+beforeAll(() => server.listen())
```

Comment:
these dont have to go in a describe block? seems like they would apply to all tests in the global space here...

Reply:
Every es6 import (module) is its own isolated context.